File: Solution/Anmol_Chawla_task2.py
import sys
from pyspark import SparkContext
from pyspark import SparkConf
from pyspark.sql import SQLContext
from pyspark.sql import Row
from pyspark.sql.functions import *


sc = SparkContext.getOrCreate()
sqlContext = SQLContext(sc)


ip_file1 = sys.argv[1]
ip_file2 = sys.argv[2]
op_file  =  sys.argv[3]


df1 = sqlContext.read.options(header='true', inferschema='true').csv(ip_file1).drop("timestamp","userId")
df2 = sqlContext.read.options(header='true', inferschema='true').csv(ip_file2).drop("timestamp","userId")



df = df1.join(df2, ['movieId'], how = "inner") \
.drop("movieId")\
.groupby('tag')\
.avg('rating')\
.sort('tag', ascending = False)\
.select(col("tag").alias("tag"),col("avg(rating)").alias("rating_avg"))\
.coalesce(1).write.options(header= "true").csv(op_file)

File: Solution/Anmol_Chawla_task2.scala
import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.sql.functions._


object task2 {
  def main(args: Array[String]) {
    val conf = new SparkConf().setMaster("local").setAppName("task2").setSparkHome("src/main/resources")
    val sc = new SparkContext(conf)
    val spark = org.apache.spark.sql.SparkSession.builder.master("local").appName("task2").getOrCreate;
    val df1 = spark.read.format("csv").option("header", "true").option("mode", "DROPMALFORMED").option("inferSchema","true").load(args(0)).drop("userId", "timestamp")
    val df2 = spark.read.format("csv").option("header", "true").option("mode", "DROPMALFORMED").option("inferSchema","true").load(args(1)).drop("userId", "timestamp")
    val df = df1.join(df2, df1("movieId") <=> df2("movieId") , "inner").drop("movieId").groupBy("tag").avg("rating").orderBy(desc("tag")).withColumnRenamed("avg(rating)", "rating_avg").coalesce(1).write.format("csv").option("header","true").save(args(2))
  }
}